src/server/api/controllers/profile.controller.js

```diff
@@ -18,6 +18,49 @@ const getStudentsProfile = async (userId) => {
     return error.message;
   }
 };
+const getMentorsProfile = async (userId) => {
+  try {
+    const profiles = await knex('users')
+      .select(
+        'users.name',
+        'users.mentor_role as mentorRole',
+        'users.timezone',
+        'organization_id',
+      )
+      .join('user_roles', 'users.id', 'user_roles.user_id')
+      .join('roles', 'user_roles.role_id', 'roles.id')
+      .join(
+        'organizations',
+        'user_organizations.organization_name',
+        'organization_name',
+      )
+      .where('users.id', userId)
+      .where('roles.name', 'mentor');
+    return profiles[0];
+  } catch (error) {
+    return error.message;
+  }
+};
+const editStudentProfile = async (body, userId) => {
+  try {
+    const studentData = await knex('users')
+      .select('users.name', 'users.timezone')
+      .leftJoin('user_roles', 'users.id', 'user_roles.user_id')
+      .leftJoin('roles', 'user_roles.role_id', 'roles.id')
+      .where('roles.name', 'student')
+      .where({ 'users.id': userId });
+    if (studentData.length === 0) {
+      throw new Error(`incorrect entry with the id of ${userId}`, 404);
+    } else {
+      await knex('users').where({ id: userId }).update({
+        name: body.name,
+        timezone: body.timezone,
+      });
+    }
+  } catch (error) {
+    return error.message;
+  }
+};
```

Comment:
This block of code needs to be removed
1. It wasn't  part of your issue
2.  was aimed to update a student's profile
3. is repeated below from line 65 to 84